Assets/Scripts/Menu Scripts/DroneButtons.cs

```diff
@@ -33,86 +27,46 @@ void OnClickEvent()
     {
         if (startMission)
         {
-            Debug.Log("Start Mission Button: Currently not connected to Drone_ROS Connection");
-
-            if (simulation)
-            {
-                GameObject world = GameObject.FindGameObjectWithTag("World");
-                DroneSimulationManager droneSim = world.GetComponent<DroneSimulationManager>();
-                droneSim.FlyNextWaypoint(true);
-                return;
-            }
-
+            Drone selectedDrone = WorldProperties.GetSelectedDrone();
```

Comment:
You can extract the selectedDrone and connection lines of code. they are common across all the buttons.

Reply:
yep did so, good catch